survey_contact_generation/models/survey_user_input.py

```diff
@@ -0,0 +1,93 @@
+            for line in basic_inputs
+        }
+        for line in elegible_inputs - basic_inputs:
+            field_name = line.question_id.res_partner_field.name
+            if line.question_id.res_partner_field.ttype == "many2one":
+                vals[
+                    field_name
+                ] = line.suggested_answer_id.res_partner_field_resource_ref.id
+            elif line.question_id.res_partner_field.ttype == "many2many":
+                vals.setdefault(field_name, [])
+                vals[field_name] += [
+                    (4, line.suggested_answer_id.res_partner_field_resource_ref.id)
+                ]
+            # We'll use the comment field to add any other infos
+            elif field_name == "comment":
+                vals.setdefault("comment", "")
+                value = (
+                    line.suggested_answer_id.value
+                    if line.answer_type == "suggestion"
+                    else line[f"value_{line.answer_type}"]
+                )
+                vals["comment"] += f"\n{line.question_id.title}: {value}"
+            # Create the parent company
+            elif field_name == "company_name" and self.survey_id.create_parent_contact:
+                if line[f"value_{line.answer_type}"]:
+                    vals["parent_id"] = (
+                        self.env["res.partner"]
+                        .create(
+                            {
+                                "name": line[f"value_{line.answer_type}"],
+                                "company_type": "company",
+                            }
+                        )
+                        .id
+                    )
+            else:
+                if line.question_id.question_type == "multiple_choice":
+                    if not vals.get(field_name):
+                        vals[field_name] = line.suggested_answer_id.value
+                    else:
+                        vals[field_name] += line.suggested_answer_id.value
+                else:
+                    vals[field_name] = line.suggested_answer_id.value
+        vals["generating_survey_user_input_id"] = self.id
+        return vals
+
+    def _create_contact_post_process(self, partner):
+        """After creating the lead send an internal message with the input link"""
+        partner.message_post_with_source(
+            "mail.message_origin_link",
+            render_values={"self": partner, "origin": self.survey_id},
+            subtype_xmlid="mail.mt_note",
+        )
+
+    def _get_existing_partner(self, email, limit=1):
+        """Hook method that can be used to change the behavior of contact generation"""
+        return self.env["res.partner"].search([("email", "=ilike", email)], limit=limit)
+
+    def _mark_done(self):
+        """Generate the contact when the survey is submitted"""
+        for user_input in self.filtered(
+            lambda r: r.survey_id.generate_contact and not self.partner_id
+        ):
```

Comment:
Sorry, but this is not working for me. I completed the survey, but the contact was not created. I'm not sure if the filter is correct. Please check my suggestion or investigate why this isn't working, or let me know if I missed any step or configuration.

```suggestion
        for user_input in self.filtered(
            lambda r: r.survey_id.generate_contact and not r.partner_id
        ):
```

![image](https://github.com/user-attachments/assets/1bebebf1-3d90-46f8-bfd7-3a17906558a6)
![image](https://github.com/user-attachments/assets/7746f712-8919-4b45-b085-fcefccccb81c)
![image](https://github.com/user-attachments/assets/a624bcc0-8169-4179-8473-39bcd1c80aec)
![image](https://github.com/user-attachments/assets/ca7efb85-b834-4051-8be1-f8ffef053571)